cetmix_tower_yaml/models/cx_tower_yaml_mixin.py

```diff
@@ -108,23 +126,27 @@ def _post_process_record_values(self, values):
             dict(): processed values
         """
         # We don't need id because we are not using it
-        values.pop("id")
+        values.pop("id", None)
 
         # Add YAML format version and model
-        model_name = self._name.replace("cx.tower.", "").replace(".", "_")
-        values.update(
-            {
+        if not self._context.get("no_yaml_service_fields"):
+            model_name = self._name.replace("cx.tower.", "").replace(".", "_")
+            model_values = {
                 "cetmix_tower_yaml_version": self.CETMIX_TOWER_YAML_VERSION,
                 "cetmix_tower_model": model_name,
             }
-        )
+        else:
+            model_values = {}
 
         # Parse access level
-        if "access_level" in values:
-            values.update(
-                {"access_level": self.TO_YAML_ACCESS_LEVEL[values["access_level"]]}
+        access_level = values.pop("access_level", None)
+        if access_level:
+            model_values.update(
+                {"access_level": self.TO_YAML_ACCESS_LEVEL[access_level]}
```

Comment:
> _🛠️ Refactor suggestion_
> 
> **Add validation for invalid access levels**
> 
> The code assumes the access level exists in `TO_YAML_ACCESS_LEVEL`. Consider adding validation to handle invalid access levels gracefully.
> 
> ```diff
>  access_level = values.pop("access_level", None)
>  if access_level:
> +    yaml_access_level = self.TO_YAML_ACCESS_LEVEL.get(access_level)
> +    if not yaml_access_level:
> +        raise ValidationError(
> +            _("Invalid access level: %(acv)s", acv=access_level)
> +        )
>      model_values.update(
> -        {"access_level": self.TO_YAML_ACCESS_LEVEL[access_level]}
> +        {"access_level": yaml_access_level}
>      )
> ```
> 
> 📝 Committable suggestion

This suggestion is not relevant because "access_level" is a Selection field so its values are always pre-defined.